enhancements/ingress/global-options-enable-hsts.md

```diff
@@ -0,0 +1,252 @@
+  {{- with $hsts := firstMatch $hstsPattern (index $cfg.Annotations "haproxy.router.openshift.io/hsts_header") }}
+    http-response set-header Strict-Transport-Security {{$hsts}}
+  {{- end }}{{/* hsts header override per-route */}}
+{{- end }}{{/* is "edge" or "reencrypt" */}}
+````
+
+#### Enable HSTS on a node-type basis (infra-only, worker-only, or both)
+(TBD) Clarification has been requested on this requirement.
+
+#### Enable HSTS for particular subdomains (TBD - dependency on wildcard DNS record/certificate)
+The`Strict-Transport-Security` header already has a directive called `includeSubDomains` which 
+will automatically apply to any subdomains of the host's domain name.  Unfortunately, once the 
+`max-age` reaches 0, it will remove the entire HSTS policy associated with the HSTS host.  Because
+of this encompassing application of the `max-age` limit to the entire HSTS policy,  it may be best
+not to use the `includeSubDomains` directive to enable HSTS for all subdomains of a host, and 
+instead to set the header individually per subdomain.  Alternatively, it may be preferred to expire
+HSTS on all child domains when the parent expires, and use `includeSubDomains` as described above.
+
+If the requirement to enable HSTS for particular subdomains isn't referring to the `includeSubDomain`
+directive, then we will use the requester clarification when it arrives. We need to know more about 
+what they mean by "subdomains of apps.example.com", and whether they are referring to the application's
+use of the wildcard DNS record/certificate. (TBD)
+
+If the latter situation is true, the HAProxy http-response line in the configuration file would need to 
+examine the host name of the route, find the configured subdomain to match against, and set the header 
+like this: 
+````
+http-response set-header Strict-Transport-Security {{$hsts}} if { isHSTSSubDomain req.hdr(host) }
+````
+The function `isHSTSSubDomain` would be specified to take a host name as input and return a boolean true
+if the host name is part of a subdomain configured for HSTS, and false if not.  This may also require
+a new field in `BackendConfig` to hold a list of subdomains to be configured for HSTS.  Details are TBD.
+
+#### Disable HSTS per-route when it is enabled globally
+As mentioned previously, the `Strict-Transport-Security` annotation called 
+`haproxy.router.openshift.io/hsts_header` has a required directive `max-age` that determines
+its lifespan.  A per-route override on globally enabled HSTS can be indicated by setting the `max-age` 
+value in the route's annotation to zero, as shown in this example:
+```
+metadata:
+  annotations:
+    haproxy.router.openshift.io/hsts_header: max-age=0     
+````
+
+### User Stories
+
+#### As a cluster administrator, I want to enable HSTS for all routes globally
+Add an environment variable to the IngressController's deployment spec, for example, setting 
+`ROUTER_DEFAULT_HSTS` to `max-age=31536000` 
```

Comment:
I think there may be some trailing spaces in this section that are throwing lint errors in the link CI job.